### PR TITLE
fix(emqx_management): handle multiple routes in topics/{topic} API

### PR DESCRIPTION
**Pull request by olcai**
The `topics/{topic}` API endpoint would return `500 - Internal Error` if a topic had multiple routes. This is now fixed by returning a list of routes.

Fixes EMQX-8679 and Github Issue #9540.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [X] Added tests for the changes
- [X] Changed lines covered in coverage report
- [X] Change log has been added to `changes/<version>/(feat|fix)-<PR-id>.en.md` and `.zh.md` files
- [X] For internal contributor: there is a jira ticket to track this change
- [ ] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [ ] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_management/test/emqx_mgmt_api_topics_SUITE.erl [hunk 1]
```diff
@@ -72,8 +72,19 @@ t_nodes_api(_) ->
...
+    ok = emqx_router:delete_route(Topic, DummyNode),
+
+    [
+        #{<<"topic">> := Topic, <<"node">> := Node1},
+        #{<<"topic">> := Topic, <<"node">> := Node2}
+    ] = emqx_json:decode(RouteResponse, [return_maps]),
+
+    DummyNodeBin = atom_to_binary(DummyNode),
+    ?assert(Node1 == Node orelse Node2 == Node),
+    ?assert(Node1 == DummyNodeBin orelse Node2 == DummyNodeBin).
```
Comment (thalesmg) [inline apps/emqx_management/test/emqx_mgmt_api_topics_SUITE.erl:90]: Tiny nit:

```suggestion
    ?assertEqual(lists:usort([Node, DummyNodeBin]), lists:usort([Node1, Node2])).
```

**Review by thalesmg — DISMISSED** [review]

**Comment by thalesmg** [comment]
Just confirming: would this be intended for `release-50` as well?

File: apps/emqx_management/test/emqx_mgmt_api_topics_SUITE.erl (hunk 1 above)
Comment (olcai) [reply apps/emqx_management/test/emqx_mgmt_api_topics_SUITE.erl]: Yep, that looks better. Changed to your suggestion.

**Comment by olcai** [comment]
> Just confirming: would this be intended for `release-50` as well?

I dunno. The bug is not a priority one, and I haven't checked if my change impacts the dashboard in any way. Will check!

File: changes/v5.0.16/fix-9824.zh.md [hunk 2]
```diff
@@ -0,0 +1 @@
+The `topics/{topic}` API endpoint would return 500 - Internal Error if a topic had multiple routes. This is fixed by returning a list of routes.
```
Comment (olcai) [inline changes/v5.0.16/fix-9824.zh.md:1]: Requires translation

File: changes/v5.0.16/fix-9824.en.md [hunk 3]
```diff
@@ -0,0 +1 @@
+The `topics/{topic}` API endpoint would return 500 - Internal Error if a topic had multiple routes. This is fixed by returning a list of routes.
```
Comment (ieQu1) [inline changes/v5.0.16/fix-9824.en.md:1]:
```suggestion
The `topics/{topic}` API endpoint would return `500 - Internal Error` if a topic had multiple routes. This is fixed by returning a list of routes.
```
?

Comment (olcai) [reply changes/v5.0.16/fix-9824.en.md]: Fixed!

File: changes/v5.0.16/fix-9824.zh.md [hunk 4]
```diff
@@ -0,0 +1 @@
+The `topics/{topic}` API endpoint would return `500 - Internal Error` if a topic had multiple routes. This is fixed by returning a list of routes.
```
Comment (lafirest) [inline changes/v5.0.16/fix-9824.zh.md:1]: 修复：当存在多个路由信息时，`topics/{topic}` 将会返回 `500 - Internal Error` 的问题，现在将会正确的返回路由信息列表。

Comment (olcai) [reply changes/v5.0.16/fix-9824.zh.md]: Thank you!

**Review by zmstone — APPROVED** [review]

**Pull request merged**